### PR TITLE
Feature/stake position table

### PR DESCRIPTION
**Pull request by paul-pelith**
image

### CODE REVIEW

File: src/components/StakingPage/StakePositionTable/TableComponent/index.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+export function expiryDateRenderer(_: unknown, record: ITableData) {
+  const isExpired = record.until.mul(1000).lt(Date.now());
+
+  const text = isExpired
+    ? 'Expired'
+    : `Left ${record.until
+        .mul(1000)
+        .sub(Date.now())
+        .div(24 * 60 * 60 * 1000)
```
Comment (ytwang00) [inline src/components/StakingPage/StakePositionTable/TableComponent/index.tsx:33]: make it a constants

Comment (ytwang00) [inline src/components/StakingPage/StakePositionTable/TableComponent/index.tsx:33]: make it a constants

File: src/components/StakingPage/index.tsx [hunk 2]
```diff
@@ -235,50 +288,22 @@ const Staking = () => {
           <hr sx={styles.hr} />
           <div sx={styles.sHakkaRewardLinkWrapper}>
             <span>Earn more Hakka</span>
-            <a sx={styles.sHakkaRewardLinkBtn} href="/farms/0xF4D1F9674c8e9f29A69DC2E6f841292e675B7977" rel="noreferrer">
+            <a
+              sx={styles.sHakkaRewardLinkBtn}
+              href="/farms/0xF4D1F9674c8e9f29A69DC2E6f841292e675B7977"
```
Comment (ytwang00) [inline src/components/StakingPage/index.tsx:293]: make this a constant

/src/constants/index.ts

Comment (ytwang00) [inline src/components/StakingPage/index.tsx:293]: make this a constant

/src/constants/index.ts

File: src/components/StakingPage/StakePositionTable/index.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+export default function StakePositionTable(props: IProps) {
+  const { data, onRedeem, onRestake } = props;
+
+  const [showArchive, setShowArchive] = useState(false);
+
+  const handleArchive = useCallback(() => {
+    setShowArchive((state) => !state);
+  }, []);
```
Comment (ytwang00) [inline src/components/StakingPage/StakePositionTable/index.tsx:24]: Don't we need to put setShowArchive in dependency?

Comment (ytwang00) [inline src/components/StakingPage/StakePositionTable/index.tsx:24]: Don't we need to put setShowArchive in dependency?

File: src/components/StakingPage/StakePositionTable/index.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+          stakedHakkaStr: (+formatUnits(raw.stakedHakka)).toFixed(4),
+          sHakkaReceivedStr: (+formatUnits(raw.sHakkaReceived)).toFixed(4),
+        };
+        if (state) {
+          nonArchiveList.push(_tmpRaw);
+        } else {
+          archiveList.push(_tmpRaw);
+        }
+      });
+    archiveList.sort((a, b) => a.until.sub(b.until).toNumber());
```
Comment (ytwang00) [inline src/components/StakingPage/StakePositionTable/index.tsx:45]: the same sorting function

Comment (ytwang00) [inline src/components/StakingPage/StakePositionTable/index.tsx:45]: same sorting function

File: src/components/StakingPage/StakePositionTable/index.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+  const handleArchive = useCallback(() => {
+    setShowArchive((state) => !state);
+  }, []);
+
+  const tableData: ITableData[] = useMemo(() => {
+    const archiveList: ITableData[] = [];
+    const nonArchiveList: ITableData[] = [];
+    data
+      .forEach((raw) => {
+        const state =
```
Comment (ytwang00) [inline src/components/StakingPage/StakePositionTable/index.tsx:31]: I think for readability reason, make this state a enum could be better

Comment (ytwang00) [inline src/components/StakingPage/StakePositionTable/index.tsx:31]: I think for readability reason, make this state a enum could be better

File: src/components/StakingPage/StakePositionTable/TableComponent/index.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import styles from '../styles';
+import images from '../../../../images';
+
+const vaultImage = [
+  images.iconVaultArchive,
+  images.iconStaking,
+  images.iconRedeem,
+];
+
+export function vaultIconsRenderer(_: unknown, record: ITableData) {
```
Comment (ytwang00) [inline src/components/StakingPage/StakePositionTable/TableComponent/index.tsx:13]: I prefer start with render like renderVaultIcons

Comment (ytwang00) [reply src/components/StakingPage/StakePositionTable/TableComponent/index.tsx:13]: And by the document, I think we should use the first value on easy case, please discuss with me

File: src/components/StakingPage/StakePositionTable/index.tsx (hunk 3 above)
Comment (paul-pelith) [reply src/components/StakingPage/StakePositionTable/index.tsx:24]: set hook callback function is working on the current state

File: src/components/StakingPage/StakePositionTable/styles.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    size: '20px',
+  },
+};
+
+const button: ThemeUIStyleObject = {
+  border: '1px solid #dae1e3',
+  borderRadius: '8px',
+  py: '12px',
+  width: '112px',
+  color: '#2DA287',
```
[the remark this answers is not in the document]
Comment (paul-pelith) [reply src/components/StakingPage/StakePositionTable/styles.tsx:20]: will be modified by using CSS variable

**Pull request merged**